aten/src/ATen/FunctionalTensorImplBase.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <ATen/Tensor.h>
+
+namespace at {
+
+struct TORCH_API FunctionalTensorImplBase : public c10::TensorImpl {
+  explicit FunctionalTensorImplBase(c10::DispatchKeySet keyset, caffe2::TypeMeta dtype, c10::Device device);
+  explicit FunctionalTensorImplBase(caffe2::TypeMeta dtype, c10::Device device);
+
```

Comment:
documentary commentssss

Reply:
There are some comments in the XLA codebase which can serve as a good basis here https://github.com/pytorch/xla/blob/master/torch_xla/csrc/view.h

Reply:
totally agreed, better docs coming soon